cmd/nerdctl/build_nix_linux.go

```diff
@@ -0,0 +1,191 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/continuity/fs"
+	"github.com/containerd/nerdctl/pkg/defaults"
+	"github.com/containerd/nerdctl/pkg/idgen"
+	"github.com/containerd/nerdctl/pkg/nixutil"
+	"github.com/opencontainers/go-digest"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newBuildNixCommand() *cobra.Command {
+	shortHelp := "Reproducible image building with Nix derivations (EXPERIMENTAL)"
+	longHelp := shortHelp + `
+Examples: https://github.com/containerd/nerdctl/tree/master/examples/nix-build
+
+WARNING: This command is experimental and its behavior is subject to change.
+`
+	cmd := &cobra.Command{
+		Use:           "build-nix",
+		Short:         shortHelp,
+		Long:          longHelp,
+		RunE:          buildNixAction,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Args:          cobra.NoArgs,
+	}
+	cmd.Flags().String("nix-image", defaults.NixImage, "Nix image")
+	cmd.Flags().StringP("file", "f", "default.nix", "Name of the Nix derivation")
+	cmd.Flags().String("iidfile", "", "Write the image ID to the file")
+	return cmd
+}
+
+func buildNixAction(cmd *cobra.Command, args []string) error {
+	logrus.Warn("This command is experimental and subject to change.")
+
+	nixImage, err := cmd.Flags().GetString("nix-image")
+	if err != nil {
+		return err
+	}
+
+	file, err := cmd.Flags().GetString("file")
+	if err != nil {
+		return err
+	}
+	if _, err := os.Stat(file); err != nil {
+		return err
+	}
+	dir := filepath.Dir(file)
```

Comment:
`nix-build` can only refer to files under `filepath.Dir(file)`? Should be documented somewhere maybe.

And the following doesn't work:

```
nerdctl build-nix -f /hello.nix
WARN[0000] This command is experimental and subject to change. 
FATA[0004] failed to read link: /proc/12988/fd/3: readlink /proc/12988/fd/3: no such file or directory 
```

`nerdctl build-nix -f /tmp/ctx/hello.nix` *do* work though.